crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -1,43 +1,40 @@
+mod command_queue;
+
 use crate::{
     bundle::Bundle,
     component::Component,
     entity::{Entities, Entity},
     world::World,
 };
 use bevy_utils::tracing::debug;
+use command_queue::CommandQueueInner;
 use std::marker::PhantomData;
 
 /// A [`World`] mutation.
 pub trait Command: Send + Sync + 'static {
-    fn write(self: Box<Self>, world: &mut World);
+    fn write(self, world: &mut World);
 }
 
 /// A queue of [`Command`]s.
 #[derive(Default)]
 pub struct CommandQueue {
```

Comment:
Why does `CommandQueue` still exist? That is, should we just use the current `CommandQueueInner` type directly as `CommandQueue`?

Reply:
I would say the only reason we should leave the two separated would be to _hide_ the implementation details of `CommandQueueInner` from the user. 

However, I don't feel strongly either way. 
Specifically, if we want to collapse these abstraction, I would just remove the current `CommandQueue` and rename `CommandQueueInner` -> `CommandQueue`.

Reply:
Does anyone have a strong preference either way?
(@DJMcNab)

Reply:
I think I'd prefer not to have the CommandQueue abstraction which is entirely a duplicating wrapper around the CommandQueueInner; the public API is identical for both of them as far as I can see.